core/transaction_test.go

```diff
@@ -44,23 +50,19 @@ func TestTransactionHash(t *testing.T) {
 		},
 	}
 
-	gw, closer := testsource.NewTestGateway(utils.MAINNET)
-	defer closer.Close()
-
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
+			gw, closer := testsource.NewTestGateway(test.network)
```

Comment:
Before there was only one network. now there are multiple networks. We can create one gateway per network and then switch. I will do that. This was just for convenience. I actually started running into problems where ports were not available while testing.

Reply:
So if we want fast tests we cannot share because it creates a race condition on the close functions (when using `t.Parallel`). If we don't use `t.Parallel` it takes 5s to run on my machine, with `t.Parallel` it takes 1s.